src/types/UiaTextRangeBase.cpp

```diff
@@ -540,6 +539,14 @@ try
         const auto& buffer = _pData->GetTextBuffer();
         const auto bufferSize = buffer.GetSize();
 
+        // TODO GH#5406: create a different UIA parent object for each TextBuffer
+        // nvaccess/nvda#11428: Ensure our endpoints are in bounds
+        // otherwise, we'll FailFast catastrophically
+        if (!bufferSize.IsInBounds(_start, true) || !bufferSize.IsInBounds(_end, true))
+        {
+            throw E_FAIL;

```

Comment:
```suggestion
            THROW_HR(E_FAIL);
```

we'll need to `THROW_HR(E_FAIL)` instead of `throw E_FAIL` actually; `THROW_HR` is a wil error handling wrapper that will make sure it gets packed up into the right kind of object.

(we can't just throw integers :smile:)

Reply:
oh duh. yeah you're right.